crates/bevy_pbr/src/environment_map/generate_from_skybox.rs

```diff
@@ -0,0 +1,603 @@
+// https://research.activision.com/publications/archives/fast-filtering-of-reflection-probes
+
+use super::{
+    filter_coefficents::{FilterCoefficentsType, FILTER_COEFFICENTS},
+    EnvironmentMapLight, DIFFUSE_CONVOLUTION_SHADER_HANDLE, DOWNSAMPLE_SHADER_HANDLE,
+    FILTER_SHADER_HANDLE,
+};
+use bevy_asset::{Assets, Handle};
+use bevy_core_pipeline::Skybox;
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::{QueryItem, Without},
+    system::{Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_reflect::Reflect;
+use bevy_render::{
+    extract_component::ExtractComponent,
+    render_asset::RenderAssets,
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode},
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+        BindGroupLayoutEntry, BindingResource, BindingType, Buffer, BufferBindingType,
+        CachedComputePipelineId, ComputePassDescriptor, ComputePipelineDescriptor, Extent3d,
+        FilterMode, PipelineCache, SamplerBindingType, SamplerDescriptor, ShaderStages, ShaderType,
+        StorageTextureAccess, TextureAspect, TextureDescriptor, TextureDimension, TextureFormat,
+        TextureSampleType, TextureUsages, TextureView, TextureViewDescriptor, TextureViewDimension,
+        UniformBuffer,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    texture::{GpuImage, Image, ImageSampler, Volume},
+};
+use bevy_utils::default;
+
+/// Automatically generate an [`EnvironmentMapLight`] from a [`Skybox`].
+///
+/// Usage:
+/// * Add this component via `GenerateEnvironmentMapLight::default()` to an entity with a [`Skybox`] component.
+/// * The first frame this component is added to the skybox entity, an [`EnvironmentMapLight`]
+/// component will be generated and added to the skybox entity.
+/// * For static (non-changing) skyboxes, remove this component 1 frame after adding it to the skybox entity to save performance.
+///
+/// This component does not work on WebGPU or WebGL2.
```

Comment:
Why doesn't it work on WebGPU?

Reply:
Rg11b10Float is not supported as a storage texture on WebGPU. I think I'll actually allow runtime selection of Rgb9e5Ufloat, Rg11b10Float, and Rgba16Float.

Reply:
Didn't do Rgb9e5Ufloat because I don't think it can be used as a storage texture, but added Rgba16Float support.